src/NumericalAlgorithms/FiniteDifference/AoWeno.hpp

```diff
@@ -0,0 +1,334 @@
+             nonlinear_weights[1] * moments_sr3_1[1] +
+             nonlinear_weights[2] * moments_sr3_2[1] +
+             nonlinear_weights[3] * moments_sr3_3[1],
+         nonlinear_weights[0] / linear_weights[0] *
+                 (moments_sr5[2] - linear_weights[1] * moments_sr3_1[2] -
+                  linear_weights[2] * moments_sr3_2[2] -
+                  linear_weights[3] * moments_sr3_3[2]) +
+             nonlinear_weights[1] * moments_sr3_1[2] +
+             nonlinear_weights[2] * moments_sr3_2[2] +
+             nonlinear_weights[3] * moments_sr3_3[2],
+         nonlinear_weights[0] / linear_weights[0] * moments_sr5[3],
+         nonlinear_weights[0] / linear_weights[0] * moments_sr5[4]}};
+
+    // The polynomial values (excluding the L_0(x)) at the lower/upper faces.
+    // The polynomials are (x in [-0.5,0.5]):
+    // L0(x)=1.0 (not in array)
+    // L1(x)=x
+    // L2(x)=x^2-1/12
+    // L3(x)=x^3 - (3/20) x
+    // L4(x)=x^4 - (3/14) x^2 + 3/560
+    const std::array polys_at_plus_half{0.5, 0.16666666666666666, 0.05,
+                                        0.014285714285714289};
+
+    return {{moments[0] - polys_at_plus_half[0] * moments[1] +
+                 polys_at_plus_half[1] * moments[2] -
+                 polys_at_plus_half[2] * moments[3] +
+                 polys_at_plus_half[3] * moments[4],
+             moments[0] + polys_at_plus_half[0] * moments[1] +
+                 polys_at_plus_half[1] * moments[2] +
+                 polys_at_plus_half[2] * moments[3] +
+                 polys_at_plus_half[3] * moments[4]}};
+  }
+
+  SPECTRE_ALWAYS_INLINE static constexpr size_t stencil_width() noexcept {
+    return 5;
+  }
+};
+}  // namespace detail
+
+/*!
+ * \ingroup FiniteDifferenceGroup
+ * \brief Performs an adaptive order (AO) WENO reconstruction using a single 5th
+ * order stencil and a 3rd-order CWENO scheme.
+ *
+ * The AO-WENO(5,3) scheme is based on the scheme presented in
+ * \cite Balsara2016780 but adjusted to do reconstruction on variables instead
+ * of fluxes. The Legendre basis functions on the domain \f$\xi\in[-1/2,1/2]\f$
+ * are given by:
+ *
+ * \f{align*}{
+ * L_0(\xi) &= 1 \\
+ * L_1(\xi) &= x \\
+ * L_2(\xi) &= \xi^2-\frac{1}{12} \\
+ * L_3(\xi) &= \xi^3 - \frac{3}{20} \xi \\
+ * L_4(\xi) &= \xi^4 - \frac{3}{14} \xi^2 + \frac{3}{560}
+ * \f}
+ *
+ * The oscillation indicators are given by
+ *
+ * \f{align*}{
+ * \beta_l = \sum_{k=1}^{p}\sum_{m=1}^{p}\sigma_{km} u_{k,l} u_{m,l}
+ * \f}
+ *
+ * where \f$p\f$ is the maximum degree of the basis function used for the
+ * stencil \f$l\f$, and
+ *
+ * \f{align*}{
+ * \sigma_{km}=\sum_{i=1}^{p}\int_{-1/2}^{1/2}
+ * \frac{d^i L_k(\xi)}{d\xi^i}\frac{d^i L_m(\xi)}{d\xi^i}d\xi
+ * \f}
+ *
+ * We write the 3rd-order reconstructed polynomial \f$P^{r3}_l(\xi)\f$ as
+ *
+ * \f{align*}{
+ * P^{r3}_l(\xi) = u_0 +u_{\xi} L_1(\xi) + u_{\xi 2} L_2(\xi)
+ * \f}
+ *
+ * For the stencil \f$S^{r3}_1\f$ we get
```

Comment:
This is the notation used in the paper. The different stencils are referred to by `S^{rN}_l` and have associated polynomials `P` and oscillation indicators `\beta`. I'd personally prefer to follow the notation in the literature

Reply:
Then how about adding `which are associated with the following stencils \f$S^{r3}_l\f$.`

Reply:
I'll change the sentence introducing the polynomials to since I think that reads easier:
```
 * We write the 3rd-order reconstructed polynomial \f$P^{r3}_l(\xi)\f$
 * associated with the stencil \f$S^{r3}_l\f$ as
```